src/main/java/htsjdk/samtools/SamReader.java

```diff
@@ -81,10 +81,10 @@ public String toString() {
             }
         }
 
-        public static Type SRA_TYPE = new TypeImpl("SRA", "sra", null);
-        public static Type CRAM_TYPE = new TypeImpl("CRAM", "cram", "crai");
-        public static Type BAM_TYPE = new TypeImpl("BAM", "bam", "bai");
-        public static Type SAM_TYPE = new TypeImpl("SAM", "sam", null);
+        public static final Type SRA_TYPE = new TypeImpl("SRA", "sra", null);
+        public static final Type CRAM_TYPE = new TypeImpl("CRAM", "cram", "crai");
+        public static final Type BAM_TYPE = new TypeImpl("BAM", "bam", "bai");
+        public static final Type SAM_TYPE = new TypeImpl("SAM", "sam", null);
```

Comment:
It looks like this could be replaced with a Java enum (assuming we don't need to allow others to subclass `SamReader.Type`).